content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md

```diff
@@ -0,0 +1,210 @@
+---
+date: '2021-09-15'
+section_menu: corda-5-dev-preview
+menu:
+  corda-5-dev-preview:
+    identifier: corda-corda-5.0-dev-preview-1-tutorial-c5-basic-cordapp-state-2
+    parent: corda-5-dev-preview-1-tutorials-building-cordapp
+    weight: 1030
+tags:
+- tutorial
+- cordapp
+title: "Write an advanced state"
+---
+
+Now that you have created a state to represent a voucher, you must create a `BoardingTicket` state to represent the ticket that the customer receives when they redeem their `MarsVoucher`. This state has some additional complexity. You will be creating this state in the same directory as `MarsVoucher` (`contracts/src/main/kotlin/net/corda/missionMars/states/`).
+
+## Create the `BoardingTicket` state
+
+The `BoardingTicket` state represents the ticket that Mars Express issues to a customer when they redeem their `MarsVoucher`. The state must include:
+
+* Trip information.
+* The company issuing the ticket. The customer can then verify which company the voucher was purchased from.
+* The current owner of the ticket. The company can then verify the identity of the customer who redeems the voucher.
```

Comment:
what does this mean? the  voucher is held by some identity, are we stating that redemption isn't possible by some identites? if so that is only possible in workflow, not contract rules

Reply:
True. I can take out the second sentences in the second and third bullet points, but I don't know what explanation to give as to why these items are needed.